api/genomics/wdl/WgsCohortExtractPrep.wdl

```diff
@@ -0,0 +1,125 @@
+version 1.0
+
+workflow WgsCohortExtract {
+
+  input {
+    File participant_ids
+    String query_project
+    String wgs_dataset
+    String wgs_extraction_cohorts_dataset
+    String wgs_extraction_destination_dataset
+    String wgs_extraction_temp_tables_dataset
+    String extraction_uuid
+  }
+
+  call CreateCohortTable {
+    input:
+      participant_ids = participant_ids,
+      wgs_dataset = wgs_dataset,
+      wgs_extraction_cohorts_dataset = wgs_extraction_cohorts_dataset,
+      query_project = query_project,
+      table_name = extraction_uuid
+  }
+
+  call CreateCohortExtractTable {
+    input:
+      cohort_uuid = extraction_uuid,
+      query_project = query_project,
+      wgs_dataset = wgs_dataset,
+      wgs_extraction_cohorts_dataset = wgs_extraction_cohorts_dataset,
+      wgs_extraction_destination_dataset = wgs_extraction_destination_dataset,
+      wgs_extraction_temp_tables_dataset = wgs_extraction_temp_tables_dataset
+  }
+
+  output {
+    String cohort_extract_table = CreateCohortExtractTable.cohort_extract_table
+  }
+
+}
+
+task CreateCohortTable {
+
+  input {
+    File participant_ids
+    String wgs_dataset
+    String wgs_extraction_cohorts_dataset
+    String query_project
+    String table_name
+  }
+  
+  command <<<
+    echo "SELECT
+      sample_id,
+      sample_name
+    FROM
+      \`~{wgs_dataset}.metadata\`
+    WHERE
+      sample_name IN " > create_cohort.sql
+
+    PARTICIPANT_IDS=$(cat ~{participant_ids} | awk '{print "\""$0"\""}' | paste -sd ",")
+    echo "($PARTICIPANT_IDS)" >> create_cohort.sql
```

Comment:
This is a security hole which would allow a user to execute arbitrary BigQuery SQL as our extraction SA.